plugin/pki/path_venafi_cert_enroll.go

```diff
@@ -168,16 +172,63 @@ func (b *backend) pathVenafiCertObtain(ctx context.Context, req *logical.Request
 	}
 
 	b.Logger().Debug("Creating Venafi client:")
-	cl, cfg, timeout, err := b.ClientVenafi(ctx, req, role)
+	// here we already filter proper "Zone" to later use with cfg.Zone
+	connector, cfg, err := b.ClientVenafi(ctx, logicalRequest, role)
 	if err != nil {
-		return logical.ErrorResponse(err.Error()), nil
+		return nil, err
+	}
+
+	reqData, err := populateReqData(data, role)
+	if err != nil {
+		return nil, err
+	}
+
+	var certId string
+	if !role.NoStore && role.StoreBy == storeByHASHstring {
+		certId = getCertIdHash(*reqData, cfg.Zone, b.Logger())
+	}
+
+	if !role.IgnoreLocalStorage && role.StorePrivateKey == true && (role.StoreBy == storeBySerialString || role.StoreBySerial == true) {
+		// if we don't receive a logic response, whenever is an error or the actual certificate found in storage
+		// means we need to issue a new one
+		logicalResp := preventReissue(b, ctx, logicalRequest, reqData, &connector, role, cfg.Zone)
+		if logicalResp != nil {
+			return logicalResp, nil
+		}
+	} else if !role.IgnoreLocalStorage && role.StorePrivateKey == true && role.StoreBy == storeByHASHstring {
+		logicalResp := preventReissueLocal(b, ctx, logicalRequest, reqData, role, certId)
+		if logicalResp != nil {
+			return logicalResp, nil
+		}
 	}
 
 	var certReq *certificate.Request
-	var reqData requestData
+	certReq, err = formRequest(*reqData, role, &connector, signCSR, b.Logger())
+	if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+
+	createCertificateRequest(b, &connector, ctx, logicalRequest, role, certReq)
```

Comment:
If we are returning error, dont we want to capture and check error ?

Reply:
Done in latest commit